routers/api/v1/repo/issue.go

```diff
@@ -55,12 +61,28 @@ func ListIssues(ctx *context.APIContext) {
 		isClosed = util.OptionalBoolFalse
 	}
 
-	issues, err := models.Issues(&models.IssuesOptions{
-		RepoIDs:  []int64{ctx.Repo.Repository.ID},
-		Page:     ctx.QueryInt("page"),
-		PageSize: setting.UI.IssuePagingNum,
-		IsClosed: isClosed,
-	})
+	var issues []*models.Issue
+
+	keyword := strings.Trim(ctx.Query("q"), " ")
+	if bytes.Contains([]byte(keyword), []byte{0x00}) {
+		keyword = ""
+	}
+	var issueIDs []int64
+	var err error
+	if len(keyword) > 0 {
+		issueIDs, err = indexer.SearchIssuesByKeyword(ctx.Repo.Repository.ID, keyword)
+	}
+
+	if len(keyword) == 0 || len(issueIDs) > 0 {
```

Comment:
It seems this line is unnecessary.

Reply:
No, it's necessary, but a comment should be added to explain it. At this point, three outcomes are possible:

1. No keywords were given
2. Keywords were given, returned no results
3. Keywords were given, returned some results.

This condition guards against condition 2, because otherwise models.Issues would just be called and would return all the issues as if no keyword was passed.

So, add comment:

```go
// We don't fetch any issue if we have keywords, but SearchIssuesByKeyword returned no issue IDs.
```

Reply:
I'll add a comment.

Reply:
@thehowl you are right, but we should avoid using another function names (it can be changed and you can not track changes in comments) in comment, just explain "why" that code is needed. IMO your last part is enough:
```
// We don't fetch issues if no issue matches our keywords
```

Reply:
@Morlinest I've added a comment explaining it.